RealmSwift/Results.swift

```diff
@@ -0,0 +1,257 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+
+// MARK: MinMaxType
+
+/// Types which can be used for min()/max().
+public protocol MinMaxType {}
+extension Double: MinMaxType {}
+extension Float: MinMaxType {}
+extension Int16: MinMaxType {}
+extension Int32: MinMaxType {}
+extension Int64: MinMaxType {}
+extension Int: MinMaxType {}
+extension NSDate: MinMaxType {}
+
+// MARK: AddableType
+
+/// Types which can be used for average()/sum().
+public protocol AddableType {}
+extension Double: AddableType {}
+extension Float: AddableType {}
+extension Int16: AddableType {}
+extension Int32: AddableType {}
+extension Int64: AddableType {}
+extension Int: AddableType {}
+
+/**
+`Results` is an auto-updating container type in Realm returned from object
+queries.
+
+Results can be queried with the same predicates as `List<T>` and you can chain queries to further
+filter query results.
+
+Results cannot be created directly.
+*/
+public final class Results<T: Object>: Printable {
```

Comment:
Should probably have a protocol that `Results` and `List` implement like `RLMCollection`.